R/data_reverse.R

```diff
@@ -114,6 +137,47 @@ reverse.factor <- function(x, range = NULL, verbose = TRUE, ...) {
   original_x <- x
 
   if (!is.null(range)) {
+    # no missing values allowed
+    if (anyNA(range)) {
+      insight::format_error("`range` is not allowed to have missing values.")
+    }
+    range_ok <- TRUE
+    # if we have a vector of length 2 for range, and more factor levels,
+    # we assume `range` indicates minimum and maximum of range values
+    if (length(range) == 2 && nlevels(droplevels(x)) > 2) {
+      if (is.numeric(range)) {
+        range <- min(range):max(range)
+      } else {
+        # if range is of length 2, and we have more than 2 number of levels,
+        # range must be numeric to indicate minima and maxima. if not, stop.
+        range_ok <- FALSE
+      }
+    }
+    if (length(range) > 2 && length(range) < nlevels(droplevels(x))) {
+      # if range has more than two values, but fewer values than number of
+      # factor levels, we cannot associate the reversed scale, so stop
+      range_ok <- FALSE
+    }
+    if (!range_ok) {
+      insight::format_error(
+        "`range` must be one of the following:",
+        "- a numeric vector of length two, indicating lowest and highest value of the required range,",
+        "- a vector (numeric or character) of values with at least as many values as number of levels in `x`,",
+        "- or `NULL`."
+      )
+    }
+    # check if no or not all old levels are in new range
+    if (verbose && !any(levels(x) %in% as.character(range))) {
```

Comment:
you can put `if (verbose)` and then another if else inside

Reply:
Yeah, I thought about this, too, but wanted to avoid doubled indention ;-) But I can change it.